interface/patient_file/front_payment.php

```diff
@@ -401,6 +401,11 @@ function calcTaxes($row, $amount)
     WHERE encounter = ?", array($payrow['encounter']));
     $frow = $facilityService->getById($tmprow['facility_id']);
 
+ if($_REQUEST['radio_type_of_payment']=='pre_payment'){
+  // if pre_payment is being made then use the main office address added by Sherwin 7/21/2017
+   $frow = sqlQuery("SELECT * FROM `facility` WHERE name != '' LIMIT 1");
+   }
```

Comment:
I had to get something going for the clinic in the mean time while I knew this would cause a hashing out of the code and take weeks to come up with a solution that would make it to the code base. 
Now, once I became aware of what was happening in the code that when the prepayment is made. It is not made against an encounter so there is no facility information to pull. Not sure if the query is executed and fails or if it does not execute because of missing data. 
But I was really sure you would have me to rethink the logic that is in this section.  Since I realized $frow was missing data, that is why I setup the query to grab the first facility in the facility table.  But I see the call you are making to the class of facility service. I will run the code. I am sure it will work.